lib/fpm/package/pacman.rb

```diff
@@ -0,0 +1,394 @@
+                      ::Dir.entries(".").reject{|entry| entry =~ /^\.{1,2}$/ }))
+      end
+    end
+  end # def output
+
+  def data_tar_flags
+    data_tar_flags = []
+    if attributes[:pacman_use_file_permissions?].nil?
+      if !attributes[:pacman_user].nil?
+        if attributes[:pacman_user] == 'root'
+          data_tar_flags += [ "--numeric-owner", "--owner", "0" ]
+        else
+          data_tar_flags += [ "--owner", attributes[:deb_user] ]
+        end
+      end
+
+      if !attributes[:pacman_group].nil?
+        if attributes[:pacman_group] == 'root'
+          data_tar_flags += [ "--numeric-owner", "--group", "0" ]
+        else
+          data_tar_flags += [ "--group", attributes[:deb_group] ]
+        end
+      end
+    end
+    return data_tar_flags
+  end # def data_tar_flags
+
+  def default_output
+    v = version
+    v = "#{epoch}:#{v}" if epoch
+    if iteration
+      "#{name}_#{v}-#{iteration}_#{architecture}.#{type}"
+    else
+      "#{name}_#{v}_#{architecture}.#{type}"
+    end
+  end # def default_output
+
+  def to_s(format=nil)
+    # Default format if nil
+    # git_1.7.9.3-1_amd64.deb
+    return super("NAME-FULLVERSION-ARCH.pkg.tar#{compression_ending}") if format.nil?
+    return super(format)
+  end # def to_s
+
+  private
+
+  def generate_mtree
+    ::Dir.chdir(build_path) do
+      cmd = "LANG=C bsdtar "
+      cmd += "-czf .MTREE "
+      cmd += "--format=mtree "
+      cmd += "--options='!all,use-set,type,uid,gid,mode,time,size,md5,sha256,link' "
+      cmd += ::Dir.entries(".").reject{|entry| entry =~ /^\.{1,2}$/ }.join(" ")
+      safesystem(cmd)
+    end
+  end # def generate_mtree
+
+  # KNOWN ISSUE:
+  # If an un-matched bracket is used in valid bash, as in
+  # `echo "{"`, this function will choke.
+  # However, to cover this case basically
+  # requires writing almost half a bash parser,
+  # and it is a very small corner case.
+  # Otherwise, this approach is very robust.
+  def gobble_function(cons,prod)
+    level = 1
+    while level > 0
+      line = cons.next
+      # Not the best, but pretty good
+      # short of writing an *actual* sh
+      # parser
+      level += line.count "{"
+      level -= line.count "}"
+      if level > 0
+        prod.push(line.rstrip())
+      else
+        fine = line.sub(/\s*[}]\s*$/, "")
+        if !(fine =~ /^\s*$/)
+          prod.push(fine.rstrip())
+        end
+      end
+    end
+  end # def gobble_function
+
+  def parse_install_script(path)
+    global_lines = []
+    look_for = Set.new(["pre_install", "post_install",
+                        "pre_upgrade", "post_upgrade",
+                        "pre_remove", "post_remove"])
+    functions = {}
+    look_for.each do |fname|
+      functions[fname] = []
+    end
+
+    open(path, "r") do |iscript|
+      lines = iscript.each
+      begin
+        while true
+          line = lines.next
+          m = /^\s*(\w+)\s*\(\s*\)\s*\{\s*([^}]+?)?\s*(\})?\s*$/.match(
+            line)
```

Comment:
put this on the line above?

Maybe put the regexp into a constant and then have `m = INSTALL_SCRIPT_LINE_REGEX.match(line)` or something?